dspace-server-webapp/src/test/java/org/dspace/app/rest/SubscriptionRestRepositoryIT.java

```diff
@@ -0,0 +1,1223 @@
+        SubscriptionParameter subscriptionParameter = new SubscriptionParameter();
+        subscriptionParameter.setName("frequency");
+        subscriptionParameter.setValue("D");
+        subscriptionParameterList.add(subscriptionParameter);
+        Subscription subscription = SubscribeBuilder.subscribeBuilder(context,
+                                    "content", collection, admin, subscriptionParameterList).build();
+        context.restoreAuthSystemState();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> newSubscription = new HashMap<>();
+        newSubscription.put("subscriptionType", "content");
+        List<Map<String, Object>> list = new ArrayList<>();
+        Map<String, Object> sub_list = new HashMap<>();
+        sub_list.put("name", "frequency");
+        sub_list.put("value", "W");
+        list.add(sub_list);
+        newSubscription.put("subscriptionParameterList", list);
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(put("/api/core/subscriptions/" + subscription.getID())
+                        .param("eperson_id", admin.getID().toString())
+                        .param("resource", collection.getID().toString())
+                        .content(objectMapper.writeValueAsString(newSubscription))
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                        .andExpect(status().isForbidden());
+    }
+
+    @Test
+    public void putSubscriptionTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        List<SubscriptionParameter> subscriptionParameterList = new ArrayList<>();
+        SubscriptionParameter subscriptionParameter = new SubscriptionParameter();
+        subscriptionParameter.setName("frequency");
+        subscriptionParameter.setValue("D");
+        subscriptionParameterList.add(subscriptionParameter);
+        Subscription subscription = SubscribeBuilder.subscribeBuilder(context,
+                                    "content", collection, eperson, subscriptionParameterList).build();
+        context.restoreAuthSystemState();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> newSubscription = new HashMap<>();
+        newSubscription.put("subscriptionType", "content");
+        List<Map<String, Object>> list = new ArrayList<>();
+        Map<String, Object> sub_list = new HashMap<>();
+        sub_list.put("name", "frequency");
+        sub_list.put("value", "W");
+        list.add(sub_list);
+        newSubscription.put("subscriptionParameterList", list);
+
+        String tokenSubscriber = getAuthToken(eperson.getEmail(), password);
+        getClient(tokenSubscriber).perform(put("/api/core/subscriptions/" + subscription.getID())
+                                  .param("resource", collection.getID().toString())
+                                  .param("eperson_id", eperson.getID().toString())
```

Comment:
The `PUT` endpoint should not accept parameters for `resource` or `eperson_id`, because you cannot change those fields.  This test appears to prove those params are OK?  We need to have a test that proves these params are either *invalid* or do not result in a modification to the subscription (e.g. passing in a different resource or a different eperson doesn't change the subscription).

Currently *every* `PUT` test has these params and they probably shouldn't be there.